include/zephyr/arch/riscv/arch_inlines.h

```diff
@@ -17,6 +17,7 @@ static ALWAYS_INLINE uint32_t arch_proc_id(void)
 
 #ifdef CONFIG_SMP
 	__asm__ volatile("csrr %0, mhartid" : "=r" (hartid));
+	hartid -= CONFIG_RV_BASE_CPU;
```

Comment:
Within a given implementation the rules for mapping between the real cpu ID and the zephyr cpu ID are generally known and I would expect that anywhere that the value is used this would be catered for.

Reply:
Hi @andyross, 
Thanks for this suggestion, I am starting to look into this. I will try include it with our next rebase

Reply:
Hi @andyross 
I have implemented two API's locally (one to get zephyr cpu id from proc_id and vice versa) and have tested them. I have put the prototypes in zephyr/include/zephyr/sys/arch_interface.h:
```
/**
 * @brief Returns Zephyr CPU ID when arch-specific CPU ID's differ.
 * 
 * In multiprocessor architectures, the low-level unique ID value
 * associated with the current CPU is not guaranteed to match
 * Zephyr's own sequential CPU ID's. This function allows the 
 * translation between an architectures CPU ID to a Zephyr
 * CPU ID.
 *
 * @return Zephyr CPU ID translated from arch-specific CPU ID.
*/

static inline uint32_t arch_proc_id_to_cpu(uint32_t proc_id);

/**
 * @brief Returns arch-specific CPU ID, given a Zephyr CPU ID.
 *
 * The inverse of arch_proc_id_to_cpu(uint32_t proc_id). This
 * function gets the architecture-specific unique CPU ID
 * from a given Zephyr CPU ID.
 *
 * @return Architecture-specific unique CPU ID/
*/

static inline uint32_t arch_cpu_to_proc_id(uint32_t cpu_id);
```
and have defined the above in zephyr/include/zephyr/arch/riscv/arch_inlines.h:
```
static ALWAYS_INLINE uint32_t arch_proc_id_to_cpu(uint32_t hartid)
{
	uint32_t cpu_id;
	cpu_id = hartid -= CONFIG_RV_BASE_CPU;
	return cpu_id;
}

static ALWAYS_INLINE uint32_t arch_cpu_to_proc_id(uint32_t cpu_id)
{
	uint32_t hartid;
	hartid = cpu_id + CONFIG_RV_BASE_CPU;
	return hartid;
}
```
I want to ask you if you think this is suitable? And also if you think it is, should I open a separate Pull Request, just for the above?
Would really appreciate any guidance on this. 
Cheers!